wgengine/netstack/netstack.go

```diff
@@ -242,14 +242,28 @@ func (ns *Impl) acceptTCP(r *tcp.ForwarderRequest) {
 
 func (ns *Impl) forwardTCP(client *gonet.TCPConn, address string) {
 	ns.logf("[v2] netstack: forwarding to address %s", address)
-	server, err := ns.dialTCP(address)
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
is there no better context here? can be subscribe to something on `client` to get notified when the peer sends a RST or something?